url5a01f2c1d9_core9_core_tgz-pJ8-io_core9_core_boot_BootstrapFrameworkJ8/src/net/xeoh/plugins/base/impl/classpath/ClassPathManager.java

```diff
@@ -0,0 +1,321 @@
+                try {
+                    final Class<?> c = Class.forName(name, false, classLoader);
+
+                    // No interfaces please
+                    if (c.isInterface()) continue;
+
+                    if (superclass.isAssignableFrom(c) && !superclass.getCanonicalName().equals(c.getCanonicalName())) {
+                        rval.add(name);
+                    }
+                } catch (ClassNotFoundException e) {
+                    this.logger.fine("ClassNotFoundException. Unable to inspect class " + name + " although it appears to be one.");
+
+                    // Print all causes, helpful for debugging
+                    Throwable cause = e.getCause();
+                    while (cause != null) {
+                        this.logger.fine("Reason " + cause.getMessage());
+                        cause = cause.getCause();
+                    }
+                } catch (final NoClassDefFoundError e) {
+                    this.logger.finer("Ignored class " + name + " due to unresolved dependencies");
+
+                    // Print all causes, helpful for debugging
+                    Throwable cause = e.getCause();
+                    while (cause != null) {
+                        this.logger.fine("Reason " + cause.getMessage());
+                        cause = cause.getCause();
+                    }
+                } catch (SecurityException e) {
+                    this.logger.fine("SecurityException while trying to find subclasses. Cause of trouble: " + name + ". This does not neccessarily mean problems however.");
+
+                    // Print all causes, helpful for debugging
+                    Throwable cause = e.getCause();
+                    while (cause != null) {
+                        this.logger.fine("Reason " + cause.getMessage());
+                        cause = cause.getCause();
+                    }
+                }
+            }
+        } catch (NoSuchRealmException e1) {
+            e1.printStackTrace();
+        }
+
+        // Update the cache information
+        if (cacheEntry != null) {
+            cacheEntry.subclasses.put(superclass.getCanonicalName(), rval);
+        }
+
+        return rval;
+    }
+
+    /**
+     * Adds a classpath location to this manager.
+     * 
+     * @param location
+     */
+    public void registerLocation(AbstractClassPathLocation location) {
+        if (!this.initializedProperly) return;
+
+        try {
+            final ClassRealm newRealm = this.classWorld.newRealm(location.getRealm(), getClass().getClassLoader());
+            final URI[] classpathLocations = location.getClasspathLocations();
+            for (URI uri : classpathLocations) {
+                newRealm.addConstituent(uri.toURL());
+            }
+
+        } catch (DuplicateRealmException e) {
+            // Happens for #classpath realms ...
+        } catch (MalformedURLException e) {
+            e.printStackTrace();
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This code contains a potential information leakage in the handling of the following *exception logging*: (MalformedURLException e). Stack trace information reveals variable names, method names, line numbers, and other parts of the code.

**Fix**
Release data more selectively, or declassify the data before it&#39;s released. 

**More info**
[Learn more about information leakage](https://cwe.mitre.org/data/definitions/497.html) (external link).